packages/ui/src/components/Visualization/Canvas/Canvas.tsx

```diff
@@ -186,21 +87,101 @@ export const Canvas: FunctionComponent<PropsWithChildren<CanvasProps>> = ({ enti
       },
     };
 
-    controller.fromModel(model, false);
-  }, [activeLayout, controller, entities, visibleFlows]);
+    requestAnimationFrame(() => {
+      controller.fromModel(model, true);
+    });
+  }, [controller, entities, visibleFlows]);
+
+  const handleSelection = useCallback((selectedIds: string[]) => {
+    setSelectedIds(selectedIds);
+  }, []);
+  useEventListener<SelectionEventListener>(SELECTION_EVENT, handleSelection);
 
+  /** Set select node and pan it into view */
   useEffect(() => {
-    const timeoutId = setTimeout(
-      action(() => {
-        controller.getGraph().fit(80);
-      }),
-      500,
-    );
+    let resizeTimeout: number | undefined;
+
+    if (!selectedIds[0]) {
+      setSelectedNode(undefined);
+    } else {
+      const selectedNode = controller.getNodeById(selectedIds[0]);
+      if (selectedNode) {
+        setSelectedNode(selectedNode as unknown as CanvasNode);
```

Comment:
Why do we need to cast to unknown before casting to another type?

Reply:
This is because the `Node` type has fewer properties compared to the `CanvasNode` type. The `CanvasNode` type inherits from `NodeModel`, and it's a different type. When a situation like this happens, Typescript asks to add a cast to `unknown` first to signal that this is not a 1:1 comparison.

Reply:
:facepalm:

Reply:
thanks for the explanation